src/proto/THORChain.proto

```diff
@@ -0,0 +1,35 @@
+syntax = "proto3";
+
+package TW.THORChain.Proto;
+option java_package = "wallet.core.jni.proto";
+
+import "Cosmos.proto";
+
+message Message {
+    oneof message_oneof {
+        Cosmos.Proto.Message.Send send_coins_message = 1;
+    }
+}
+
+// Input data necessary to create a signed order.
+message SigningInput {
```

Comment:
any chance we can reuse Cosmos proto? just pass different message type

Reply:
It's partly reused, the top-level entities are duplicated in ThorChain proto, that's the best solution I found so far.
A special Thor send message inside Cosmos is a good idea, though the we have Thorchain specific deviations inside the Cosmos implementation.